services/src/modules/directives/error-handler.ts

```diff
@@ -0,0 +1,125 @@
+import { gql } from 'apollo-server-core';
+import { GraphQLFieldResolverParams } from 'apollo-server-types';
+import { defaultFieldResolver, FieldNode, GraphQLField, GraphQLObjectType, GraphQLResolveInfo } from 'graphql';
+import { SchemaDirectiveVisitor } from 'graphql-tools';
+import { injectArgs } from '../arguments-injection';
+import { RequestContext } from '../context';
+
+const defaultObjectResolver = (
+  source: unknown,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  _fields: Record<string, ReadonlyArray<FieldNode>>,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  _context: RequestContext,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  _info: GraphQLResolveInfo
+) => source;
+
+export class ErrorHandlerDirective extends SchemaDirectiveVisitor {
+  visitObject(object: GraphQLObjectType<unknown, RequestContext>) {
+    const originalResolve = object.resolveObject || defaultObjectResolver;
+    const { catchError, throwError } = this.args as ErrorHandlerDirectiveArgs;
+
+    object.resolveObject = async (source, fields, context, info) => {
+      const injectionArgs = { source, args: {}, context, info };
+      let result;
+      try {
+        result = await originalResolve.call(object, source, fields, context, info);
+      } catch (error) {
+        result = handleCatchError(catchError, error, injectionArgs, result);
+      }
+      throwError && handleThrowError(throwError, result, injectionArgs);
```

Comment:
`handleThrowError` is a little confusing here (since it doesn't always throw an error), maybe something like `handleThrowErrorCondition`